### PR TITLE
chore(deps): update multiple deps

### PR DESCRIPTION
**Pull request by kavitha186**

### CODE REVIEW

File: package-lock.json
Comment (aorinevo) [inline package-lock.json:1]: No need to commit package-lock since we have yarn.lock.

**Review by aorinevo — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (kavitha186) [reply package-lock.json:1]: yes fixed

**Review by aorinevo — APPROVED** [review]

**Comment by aorinevo** [comment]
We can drop support for Node v12 and v14.

**Comment by aorinevo** [comment]
Opened a separate PR to update CI and Release workflows (see https://github.com/NerdWalletOSS/shepherd/pull/605 )

**Comment by aorinevo** [comment]
Closing this in favor of https://github.com/NerdWalletOSS/shepherd/pull/605 

The dependency management was non-trivial and required some review subtle changes when upgrading multiple packages.

**Pull request closed without merging**